back/engines/free/user_confirmation/app/interactors/user_confirmation/schedule_code_expiration.rb

```diff
@@ -0,0 +1,14 @@
+module UserConfirmation
+  class ScheduleCodeExpiration < ApplicationInteractor
+    delegate :user, to: :context
+
+    def call
+      schedule_code_expiration_job
+    end
+
+    def schedule_code_expiration_job
+      expire_at = user.email_confirmation_code_sent_at || Time.zone.now + 1.day
+      ExpireConfirmationCodeJob.set(wait_until: expire_at).perform_later(user, user.email_confirmation_code)
```

Comment:
This is great!!! :D Assuming this does not have a negative effect on performance, I think it would be good to apply `wait_until` to other places (mainly thinking of the periodic activities in the ActivityService - like the invite reminder). 

I guess we can safely assume that `expire_at` won't get changed through any business logic (except perhaps when changing timezone, but there it really seems acceptable)?

Reply:
However, if for some reason the job would not get executed (for example, the queue got cleared because we had production issues where jobs keep getting retried forever) then the confirmation_code remains valid forever? I'm thinking maybe it would be good to also check the expiration date when email confirmation is being performed?

Reply:
Could be useful indeed, Ill add that check too 👍